proteus/NonlinearSolvers.py

```diff
@@ -755,6 +755,8 @@ def solve(self,u,r=None,b=None,par_u=None,par_r=None):
         self.F.secondCallCalculateResidual = 0
         self.computeResidual(u,r,b)
         u[:] = r
+        if par_u is not None:
```

Comment:
was this the magic call that got parallelization to work?

Reply:
Yes, this was one of the fixes. Wasn't the main fix however.